controller/engine_controller.go

```diff
@@ -868,10 +870,22 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 	}
 
 	defer func() {
+		if err != nil {
+			return
+		}
 		engine.Status.RestoreStatus = rsMap
 		if !reflect.DeepEqual(existingEngine.Status, engine.Status) {
-			if _, updateErr := m.ds.UpdateEngineStatus(engine); updateErr != nil {
+			e, updateErr := m.ds.UpdateEngineStatus(engine)
+			if updateErr != nil {
 				err = errors.Wrapf(err, "engine monitor: Failed to update the status for engine %v: %v", engine.Name, updateErr)
+				return
+			}
+			engine = e
+		}
+		if needToSyncSnapshotCRs {
+			if syncErr := m.syncSnapshotCRs(engine); syncErr != nil {
```

Comment:
IIUC, there are two ways to create the snapshot CRs. One is create the snapshot CRs by kubectl. Another one is create the snapshot by API, and then the snapshot CR is created in this logic. Right?

If yes, might the logic be removed if the snapshot CRD is mature in the next version?

Reply:
I see. Thank you for the clarification.